# Assignment: 1 - Encoding
### Name: Lilith Bertens
### Hva ID: Bertenl

# Exercise 1A
Encode the following numbers in the basis encoding pattern: 
2, 3, 5, 7, 11, 13, 17, 19, 23.

### Answer
Sure sure, first ill convert it into binary.


In [2]:
array = {2, 3, 5, 7, 11, 13, 17, 19, 23}
Output = ""
for i in array:
    print(bin(i)[2:])

10
11
101
111
1011
1101
10001
10011
10111


Then writing this as $\ket{}$ will give us the number encoded in base.     
So the numbers encoded in a basic manner are:
$\ket{10}$, $\ket{11}$, $\ket{101}$, $\ket{111}$, $\ket{1011}$, $\ket{1101}$, $\ket{10001}$, $\ket{10011}$, $\ket{10111}$    
This is done according to:   
![](img/basis_encoding_solution.png)

# Exercise 1B
How would you encode a negative number?

### Answer
You would use the first bit as a signed bit. If it's one, the number is negative.